packages/tasit-action/src/TasitAction.test.js

```diff
@@ -25,57 +35,40 @@ describe("Contract", () => {
 
     simpleStorage = new Contract(contractAddress, contractABI);
     expect(simpleStorage).to.exist;
-    expect(simpleStorage.address).to.equal(contractAddress);
+    expect(simpleStorage.getAddress()).to.equal(contractAddress);
     expect(simpleStorage.getValue).to.exist;
     expect(simpleStorage.setValue).to.exist;
-    // Events are not implemented yet
-    //expect(simpleStorage.ValueChanged).to.exist;
   });
 
   describe("should throw error when instantiated with invalid args", () => {
     it("constructor without address and ABI", async () => {
-      try {
+      expect(() => {
         new Contract();
-        assert(false);
-      } catch (e) {
-        assert(true);
-      }
+      }).to.throw();
     });
 
     it("constructor without ABI", async () => {
-      try {
+      expect(() => {
         new Contract(contractAddress);
-        assert(false);
-      } catch (e) {
-        assert(true);
-      }
+      }).to.throw();
     });
 
     it("constructor with invalid address", async () => {
-      try {
+      expect(() => {
         new Contract("invalid address");
-        assert(false);
-      } catch (e) {
-        assert(true);
-      }
+      }).to.throw();
     });
 
     it("constructor with invalid address and valid ABI", async () => {
-      try {
+      expect(() => {
         new Contract("invalid address", contractABI);
-        assert(false);
-      } catch (e) {
-        assert(true);
-      }
+      }).to.throw();
     });
 
     it("constructor with valid address and invalid ABI", async () => {
-      try {
+      expect(() => {
         new Contract(contractAddress, "invalid abi");
```

Comment:
I've taken a look at web3.js, ethers.js and searched about `abi` validation and I didn't found an `abi` validation utils. 
Should we implement ours? We can check fieldNames.
Maybe it isn't a good idea because it could break with new versions of solidity.

Reply:
Let's flag it as a possible TODO with a GitHub issue? I think it could only break with breaking changes of Solidity (6.x), right?

Reply:
For now, what constitutes a valid ABI as far as `Contract()` is concerned? Just that it's an object and not a string?

Reply:
```javascript
static isABI = abi => {
    return abi && Array.isArray(abi);
  };
```

Reply:
https://github.com/tasitlabs/TasitSDK/issues/99

Reply:
I'll close this thread since there's a GitHub issue for it now